### DISCUSSION TITLE
Best Practice loading / watch long drift list in ListView

### DISCUSSION THREAD

**Discussion by FantaMagier** [simolus3/drift#3075] [Q&A]
Hey guys, 
I have a problem on Android that when I get a stream from Drift and I want to display it in a ListView builder. I think it's more a Flutter general problem, everything should work with Drift :D

My problem is that if you scroll down in a ListView on Android for a long time and then scroll up again only briefly, the ListView is directly at the top again although you have scrolled very far down. That's why I wanted to ask how you do it with long ListViews and how you display them correctly?

I also don't want to use shrinkWrap then I load everything directly again.

Example the following stream: 

```dart
  Stream<List<ProductEntry>> watchAllProductsSortedByNameAsc() {
    return (select(db.productEntries)
          ..where((tbl) => tbl.isHistoryData.equals(true))
          ..orderBy([(t) => OrderingTerm.asc(t.name)]))
        .watch();
  }
```

And then I simply use the AsyncValue with the .when function and display the ListView.builder as normal with the data from the snapshot.

**Comment by simolus3** [comment]
To make sure it's not a drift issue after all, you could log the length of stream updates you're getting - if there's an event with a broken list of product entries, that could impact the UI as well.

Unfortunately I'm not that familiar with Flutter so I'm not sure about what to do about this, but if it turns out to not be a drift problem you could try again with a static list view (e.g. not updated by a stream) and see if the same problem happens. If it does, it may be a Flutter issue.
I don't think this is related to that particular problem, but you are setting keys on the individual widgets in the list right?

**Reply by FantaMagier** [reply]
Hi thanks for the answer. I don't think it's a problem with drift but rather with flutter. There are also a lot of problems with the Flutter SDk on this topic. That's why I wanted to ask about experiences with big list streams. I use dismissiable ListTile and give it a UniqueKey() over each.
